src/main/java/org/embulk/input/marketo/rest/MarketoRestClient.java

```diff
@@ -0,0 +1,335 @@
+            .put("float", Types.DOUBLE)
+            .put("integer", Types.LONG)
+            .put("formula", Types.STRING)
+            .put("percent", Types.LONG)
+            .put("url", Types.STRING)
+            .put("phone", Types.STRING)
+            .put("textarea", Types.STRING)
+            .put("text", Types.STRING)
+            .put("string", Types.STRING)
+            .put("score", Types.LONG)
+            .put("boolean", Types.BOOLEAN)
+            .put("currency", Types.DOUBLE)
+            .put("date", Types.TIMESTAMP)
+            .put("reference", Types.STRING)
+            .build();
+
+    public interface PluginTask extends Task
+    {
+        @Config("endpoint")
+        String getEndpoint();
+
+        @Config("identity_endpoint")
+        String getIdentityEndpoint();
+
+        @Config("client_secret")
+        String getClientSecret();
+
+        @Config("client_id")
+        String getClientId();
+
+        @Config("marketo_limit_interval_milis")
+        @ConfigDefault("20000")
+        Integer getMarketoLimitIntervalMilis();
+
+        @Config("batch_size")
+        @ConfigDefault("300")
+        Integer getBatchSize();
+
+
+    }
+
+    public MarketoRestClient(PluginTask task, Jetty92RetryHelper retryHelper)
+    {
+        this(task.getEndpoint(), task.getIdentityEndpoint(), task.getClientId(), task.getClientSecret(), task.getBatchSize(), task.getMarketoLimitIntervalMilis(), retryHelper);
+    }
+
+    public MarketoRestClient(String endPoint, String identityEndPoint, String clientId, String clientSecret, Integer batchSize, int marketoLimitIntervalMilis, Jetty92RetryHelper retryHelper)
+    {
+        super(identityEndPoint, clientId, clientSecret, marketoLimitIntervalMilis, retryHelper);
+        this.endPoint = endPoint;
+        this.batchSize = batchSize;
+    }
+
+    public List<Column> describeLead()
+    {
+        MarketoResponse<ObjectNode> jsonResponse = doGet(endPoint + MarketoRESTEndpoint.DESCRIBE_LEAD.getEndpoint(), null, null, new MarketoResponseJetty92EntityReader<ObjectNode>(READ_TIMEOUT_MILLIS));
+        List<Column> columnsList = new ArrayList<>();
+        List<ObjectNode> fields = jsonResponse.getResult();
+        for (int i = 0; i < fields.size(); i++) {
+            ObjectNode field = fields.get(i);
+            String dataType = field.get("dataType").asText();
+            ObjectNode restField = (ObjectNode) field.get("rest");
+            String name = restField.get("name").asText();
+            columnsList.add(new Column(i, name, getType(dataType)));
+        }
+        return columnsList;
+    }
+
+    private Type getType(String dataType)
+    {
+        return TYPE_MAPPING.containsKey(dataType.toLowerCase()) ? TYPE_MAPPING.get(dataType.toLowerCase()) : Types.STRING;
+    }
+
+    public String createLeadBulkExtract(Date startTime, Date endTime, List<String> extractFields)
+    {
+        SimpleDateFormat timeFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```

Comment:
Would be better to keep this date format literal as constant because it is used in several places

Reply:
Fixed